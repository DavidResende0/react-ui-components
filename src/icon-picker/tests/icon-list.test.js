import React, { cloneElement } from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import IconList from '../icon-list';

describe('IconList test', () => {
  let intialProps;
  let styleTag;

  const TableWrapper = ({ children, ...props }) => (
    <table>
      <tbody>
        {cloneElement(children, { ...props })}
      </tbody>
    </table>
  );

  beforeEach(() => {
    intialProps = {
      activeIcon: 'crosshairs',
      type: 'fa',
      iconChanged: jest.fn(),
    };
  });

  beforeAll(() => {
    styleTag = document.createElement('style');
    styleTag.innerHTML = `
      .fa-crosshairs::before {
        content: "+";
      }
      .fa-foo::before {
        content: "+";
      }
      .fa-foo {}
      .should-not-be-picked {}`;
    document.head.appendChild(styleTag);
  });

  afterAll(() => {
    document.removeChild(styleTag);
  });

  it('should render correctly in visible state', () => {
    const wrapper = shallow(<TableWrapper><IconList {...intialProps} isVisible /></TableWrapper>).find(IconList).dive();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const wrapper = shallow(<TableWrapper><IconList {...intialProps} /></TableWrapper>).find(IconList).dive();
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should update state after isVisible change', () => {
    const wrapper = mount(<TableWrapper><IconList {...intialProps} /></TableWrapper>);
    expect(wrapper.find(IconList).state()).toEqual({ icons: undefined });
    wrapper.setProps({ isVisible: true });
    expect(wrapper.find(IconList).state()).toEqual({ icons: [['fa-crosshairs', 'fa-foo']] });
  });

  it('should call onClick correctly', () => {
    const iconChanged = jest.fn();
    const wrapper = mount(<TableWrapper><IconList {...intialProps} isVisible iconChanged={iconChanged} /></TableWrapper>);
    wrapper.find('td').first().simulate('click');

    expect(iconChanged).toHaveBeenCalledWith('fa-crosshairs');
  });
});
