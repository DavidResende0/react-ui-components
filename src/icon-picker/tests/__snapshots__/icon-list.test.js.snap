// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconList test should render correctly 1`] = `""`;

exports[`IconList test should render correctly in visible state 1`] = `
<tr
  key="fa-crosshairs,fa-foo"
>
  <td
    className="icon-list-cell "
    key="fa-crosshairs"
    onClick={[Function]}
  >
    <Memo()
      icon="fa-crosshairs"
      type="fa"
    />
  </td>
  <td
    className="icon-list-cell "
    key="fa-foo"
    onClick={[Function]}
  >
    <Memo()
      icon="fa-foo"
      type="fa"
    />
  </td>
</tr>
`;
